src/controllers/user.js

```diff
@@ -32,24 +32,28 @@ export const create = async (req, res) => {
 
 export const getById = async (req, res) => {
   const id = parseInt(req.params.id)
+  const { role } = req.user
 
   try {
-    const foundUser = await User.findById(id)
-
+    let foundUser = {}
+    const isTeacher = 'TEACHER'
+    if (role === isTeacher) {
+      foundUser = await findByUniqueAsATeacher(id)
+    } else {
+      foundUser = await User.findById(id, role)
+    }
     if (!foundUser) {
       return sendDataResponse(res, 404, { error: ERR.USER_NOT_FOUND })
     }
-
-    return sendDataResponse(res, 200, foundUser)
+    return res.status(200).json({ status: 'success', data: foundUser })
```

Comment:
use the `sendDataResponse` function when returning the response body, see line 46 for what parameters are passed into the function